tests/test_wave_01.py

```
@@ -165,3 +177,18 @@ def test_does_nothing_if_movie_not_in_watchlist():
     assert len(updated_data["watched"]) == 1
     assert movie_to_watch not in updated_data["watchlist"]
     assert movie_to_watch not in updated_data["watched"]
+
+def test_watch_movie_returns_list_if_key_error():
```

You can certainly make this decision, but personally, I'd prefer the KeyError to happen if I pass in a data structure not matching the expected data layout. Like how did this data get into the system? Especially if we're expecting data to be created in some normalized way, then the sooner I get a crash, the sooner I can track down what went wrong.

We can always catch an error further up (if this were part of a larger system), but silent failures make me very uncomfortable. It's not necessarily the responsibility of one area of our code to clean up if another hasn't initialized things properly. No one right answer, but I tend to plough ahead and let the chips fall where they may!